#! /usr/bin/env python
PACKAGE='sequoia_lane_detection'

from dynamic_reconfigure.msg import SensorLevels
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name                Type      Level     Description                   Default   Min     Max
gen.add("val_thres",        int_t,    0,        "Thres",                      175,      0,      255)
gen.add("erode_size",       int_t,    0,        "Erosion",                    1,        1,      25)
gen.add("hough_rho_res",    int_t,    0,        "Rho resolution (pix)",       10,       1,      500)
gen.add("hough_theta_res",  double_t, 0,        "Theta resolution (rad)",     0.05,     0.01,   1.0)
gen.add("hough_threshold",  int_t,    0,        "Accumulator threshold",      10,       1,      200)
gen.add("hough_min_length", int_t,    0,        "Minimum line length (pix)",  75,       5,      500)
gen.add("hough_max_gap",    int_t,    0,        "Max gap",                    10,       5,      500)

exit(gen.generate(PACKAGE, PACKAGE, "ValThresh"))
